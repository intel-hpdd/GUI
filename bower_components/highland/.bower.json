{
  "name": "highland",
  "homepage": "https://github.com/caolan/highland",
  "authors": [
    "Caolan McMahon"
  ],
  "description": "The high-level streams library.",
  "main": "dist/highland.js",
  "keywords": [
    "streams",
    "async",
    "events",
    "pipe"
  ],
  "license": "Apache 2.0",
  "ignore": [
    "**/.*",
    "Contributing.md",
    "Gruntfile.js",
    "bench",
    "docs",
    "node_modules",
    "bower_components",
    "tasks",
    "test.js",
    "test"
  ],
  "version": "2.7.1",
  "_release": "2.7.1",
  "_resolution": {
    "type": "version",
    "tag": "2.7.1",
    "commit": "197822ee483f1e38fd410863d0edb4ad07d06d4d"
  },
  "_source": "https://github.com/caolan/highland.git",
  "_target": "2.7.1",
  "_originalSource": "highland"
}