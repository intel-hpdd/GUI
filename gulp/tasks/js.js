//
// INTEL CONFIDENTIAL
//
// Copyright 2013-2016 Intel Corporation All Rights Reserved.
//
// The source code contained or described herein and all documents related
// to the source code ("Material") are owned by Intel Corporation or its
// suppliers or licensors. Title to the Material remains with Intel Corporation
// or its suppliers and licensors. The Material contains trade secrets and
// proprietary and confidential information of Intel or its suppliers and
// licensors. The Material is protected by worldwide copyright and trade secret
// laws and treaty provisions. No part of the Material may be used, copied,
// reproduced, modified, published, uploaded, posted, transmitted, distributed,
// or disclosed in any way without Intel's prior express written permission.
//
// No license under any patent, copyright, trade secret or other intellectual
// property right is granted to or conferred upon you by disclosure or delivery
// of the Materials, either expressly, by implication, inducement, estoppel or
// otherwise. Any license under such intellectual property rights must be
// express and approved by Intel in writing.

'use strict';

var paths = require('../paths.json');
var gulp = require('gulp');
var babel = require('gulp-babel');
var sourcemaps = require('gulp-sourcemaps');
var destDir = require('../dest-dir');
var relativeSourcemapsSource = require('gulp-relative-sourcemaps-source');

function jsDeps (fn) {
  return gulp.src(paths.js.deps, {
    since: gulp.lastRun(fn),
    base: '.'
  });
}

exports.jsDepsDev = function jsDepsDev () {
  return jsDeps(jsDepsDev)
  .pipe(gulp.dest('./dest'))
  .pipe(gulp.symlink('static/chroma_ui', { cwd: destDir }));
};

exports.jsDepsProd = function jsDepsProd () {
  return jsDeps(jsDepsProd)
  .pipe(gulp.dest('./dest'));
};

function socketWorker (fn) {
  return gulp.src(paths.js.socketWorker, {
    since: gulp.lastRun(fn)
  })
  .pipe(sourcemaps.init({ loadMaps: true }))
  .pipe(sourcemaps.write('.'));
}

exports.socketWorkerDev = function socketWorkerDev () {
  return socketWorker(socketWorkerDev)
  .pipe(gulp.dest('./dest'))
  .pipe(gulp.symlink('static/chroma_ui', { cwd: destDir }));
};

exports.socketWorkerProd = function socketWorkerProd () {
  return socketWorker(socketWorkerProd)
  .pipe(gulp.dest('./dest'))
  .pipe(gulp.dest('./dist'));
};

function systemConfig (fn) {
  return gulp.src(paths.js.config, {
    since: gulp.lastRun(fn),
    base: '.'
  });
}

exports.systemConfigDev = function systemConfigDev () {
  return systemConfig(systemConfigDev)
  .pipe(gulp.dest('./dest'))
  .pipe(gulp.symlink('static/chroma_ui', { cwd: destDir }));
};

var babelDev = babel.bind(null, {
  presets: [],
  babelrc: false,
  plugins: [
    'check-es2015-constants',
    'transform-flow-strip-types',
    'transform-class-properties',
    'transform-object-rest-spread',
    'transform-async-to-generator',
    'transform-es2015-classes',
    'transform-es2015-modules-systemjs'
  ]
});

var babelProd = babel.bind(null, {
  presets: ['es2015'],
  babelrc: false,
  plugins: [
    'transform-flow-strip-types',
    'transform-class-properties',
    'transform-object-rest-spread',
    [
      'angularjs-annotate',
      {
        explicitOnly: true
      }
    ],
    'transform-es2015-modules-systemjs'
  ]
});

function jsSource (fn) {
  return gulp.src(paths.js.source, {
    since: gulp.lastRun(fn),
    base: '.'
  })
  .pipe(sourcemaps.init());
}

exports.jsSourceDev = function jsSourceDev () {
  return jsSource(jsSourceDev)
  .pipe(babelDev())
  .pipe(sourcemaps.write({ sourceRoot: '/static/chroma_ui' }))
  .pipe(gulp.dest('./dest'))
  .pipe(gulp.symlink('static/chroma_ui', { cwd: destDir }));
};

exports.jsSourceProd = function jsSourceProd () {
  return jsSource(jsSourceProd)
  .pipe(babelProd())
  .pipe(relativeSourcemapsSource({dest: 'dest'}))
  .pipe(sourcemaps.write({ sourceRoot: '' }))
  .pipe(gulp.dest('./dest'));
};

exports.jsTest = function jsTest () {
  return gulp.src(paths.js.tests, {
    since: gulp.lastRun(jsTest),
    base: '.'
  })
  .pipe(sourcemaps.init())
  .pipe(babelDev())
  .pipe(sourcemaps.write({ sourceRoot: '' }))
  .pipe(gulp.dest('./dest'));
};

exports.jsTestFixtures = function jsTestFixtures () {
  return gulp.src(paths.js.testFixtures, {
    since: gulp.lastRun(jsTestFixtures),
    base: '.'
  })
  .pipe(gulp.dest('./dest'));
};

exports.jsTestDeps = function jsTestDeps () {
  return gulp.src(paths.js.testDeps, {
    since: gulp.lastRun(jsTestDeps),
    base: '.'
  })
  .pipe(gulp.dest('./dest'));
};
