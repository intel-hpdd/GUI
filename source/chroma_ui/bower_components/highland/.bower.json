{
  "name": "highland",
  "version": "2.4.0",
  "homepage": "https://github.com/caolan/highland",
  "authors": [
    "Caolan McMahon"
  ],
  "description": "The high-level streams library.",
  "main": "dist/highland.js",
  "keywords": [
    "streams",
    "async",
    "events",
    "pipe"
  ],
  "license": "Apache 2.0",
  "ignore": [
    "**/.*",
    "Contributing.md",
    "Gruntfile.js",
    "bench",
    "docs",
    "node_modules",
    "bower_components",
    "tasks",
    "test.js",
    "test"
  ],
  "_release": "2.4.0",
  "_resolution": {
    "type": "version",
    "tag": "2.4.0",
    "commit": "7ff0deb4abec773009324072d8855f394cc93467"
  },
  "_source": "https://github.com/caolan/highland.git",
  "_target": "2.4.0",
  "_originalSource": "highland"
}