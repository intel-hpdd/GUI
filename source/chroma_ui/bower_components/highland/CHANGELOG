Changelog
=========

This file does not aim to be comprehensive (you have git history for that),
rather it lists changes that might impact your own code as a consumer of
this library.

2.0.0
-----

* The `source.merge()` algorithm now evaluates the entire source stream before
  reading from all of the resulting streams in parallel (previously it would
  start reading as soon as the source emitted the next stream)
* The `merge()` function now attempts to balance inputs more fairly. For example,
  if stream A has 100 values buffered and stream B gets a new value after 100ms,
  if we read at 200ms we'll get a value from each stream. Previously it would
  exhaust the stream A buffer before reading from stream B.
