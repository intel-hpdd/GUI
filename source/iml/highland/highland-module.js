// @flow

//
// Copyright (c) 2018 DDN. All rights reserved.
// Use of this source code is governed by a MIT-style
// license that can be found in the LICENSE file.

import angular from 'angular';
import highland from 'highland';

export default angular
  .module('highland', [])
  .value('Î»', highland)
  .value('highland', highland).name;
