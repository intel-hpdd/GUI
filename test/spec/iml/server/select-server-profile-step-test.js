import serverModule from '../../../../source/iml/server/server-module';
import transformedHostProfileFixture
  from '../../../data-fixtures/transformed-host-profile-fixture.json!json';
import highland from 'highland';
import * as fp from 'intel-fp';

import { mock, resetAll } from '../../../system-mock.js';

describe('select server profile', () => {
  let SelectServerProfileStepCtrl, selectServerProfileStep;

  beforeEach(module(serverModule));

  beforeEachAsync(async function() {
    const mod = await mock('source/iml/server/select-server-profile-step.js', {
      'source/iml/server/assets/html/select-server-profile-step.html!text': {
        default: 'serverProfileStepTemplate'
      }
    });

    SelectServerProfileStepCtrl = mod.SelectServerProfileStepCtrl;
    selectServerProfileStep = mod.selectServerProfileStep();
  });

  afterEach(resetAll);

  describe('select server profile step ctrl', () => {
    let $scope,
      $stepInstance,
      data,
      createHostProfiles,
      hostProfileStream,
      selectServerProfileStep;

    beforeEach(
      inject(($rootScope, $controller) => {
        $scope = $rootScope.$new();
        $stepInstance = {
          transition: jasmine.createSpy('transition')
        };

        hostProfileStream = highland();
        spyOn(hostProfileStream, 'destroy');

        createHostProfiles = jasmine
          .createSpy('createHostProfiles')
          .and.returnValue(highland());

        data = { pdsh: 'storage0.localdomain' };

        selectServerProfileStep = $controller(SelectServerProfileStepCtrl, {
          $scope,
          $stepInstance,
          data,
          hostProfileStream,
          createHostProfiles
        });
      })
    );

    it('should setup the controller', () => {
      const instance = window.extendWithConstructor(
        SelectServerProfileStepCtrl,
        {
          pdsh: data.pdsh,
          transition: jasmine.any(Function),
          onSelected: jasmine.any(Function),
          getHostPath: jasmine.any(Function),
          pdshUpdate: jasmine.any(Function),
          close: jasmine.any(Function)
        }
      );

      expect(selectServerProfileStep).toEqual(instance);
    });

    describe('transition', () => {
      let action;
      beforeEach(() => {
        action = 'previous';
        selectServerProfileStep.transition(action);
      });

      it('should end the hostProfileSpark', () => {
        expect(hostProfileStream.destroy).toHaveBeenCalledOnce();
      });

      it('should call transition on the step instance', () => {
        expect($stepInstance.transition).toHaveBeenCalledOnceWith(action, {
          data: data
        });
      });
    });

    describe('receiving data change on hostProfileSpark', () => {
      beforeEach(
        inject(() => {
          hostProfileStream.write(transformedHostProfileFixture);
        })
      );

      it('should set the profiles', () => {
        expect(selectServerProfileStep.profiles).toEqual(
          transformedHostProfileFixture
        );
      });

      describe('receiving more data', () => {
        beforeEach(() => {
          hostProfileStream.write(transformedHostProfileFixture);
        });

        it('should keep the same profile', () => {
          expect(selectServerProfileStep.profile).toEqual(
            transformedHostProfileFixture[0]
          );
        });
      });

      describe('onSelected', () => {
        beforeEach(() => {
          selectServerProfileStep.onSelected(transformedHostProfileFixture[0]);
        });

        it('should set overridden to false', () => {
          expect(selectServerProfileStep.overridden).toEqual(false);
        });

        it('should set the profile on the scope', () => {
          expect(selectServerProfileStep.profile).toEqual(
            transformedHostProfileFixture[0]
          );
        });
      });
    });

    describe('get host path', () => {
      let item;
      beforeEach(() => {
        item = {
          address: 'address'
        };
      });

      it('should retrieve the host address', () => {
        expect(selectServerProfileStep.getHostPath(item)).toEqual(item.address);
      });
    });

    describe('pdsh update', () => {
      let pdsh, hostnames, hostnamesHash;
      beforeEach(() => {
        pdsh = 'test[001-002].localdomain';
        hostnames = ['test001.localdomain', 'test002.localdomain'];
        hostnamesHash = {
          'test001.localdomain': 1,
          'test002.localdomain': 1
        };
      });

      beforeEach(() => {
        selectServerProfileStep.pdshUpdate(pdsh, hostnames, hostnamesHash);
      });

      it('should have the hostnamesHash', () => {
        expect(selectServerProfileStep.hostnamesHash).toEqual(hostnamesHash);
      });
    });
  });

  describe('selectServerProfileStep', () => {
    it('should contain the appropriate properties', () => {
      expect(selectServerProfileStep).toEqual({
        template: 'serverProfileStepTemplate',
        controller: 'SelectServerProfileStepCtrl as selectServerProfile',
        onEnter: jasmine.any(Function),
        transition: jasmine.any(Function)
      });
    });

    it('should transition to the server status step', () => {
      const steps = {
        serverStatusStep: {}
      };

      expect(selectServerProfileStep.transition(steps)).toEqual(
        steps.serverStatusStep
      );
    });

    describe('on enter', () => {
      let onEnter,
        data,
        createOrUpdateHostsStream,
        getHostProfiles,
        waitForCommandCompletion,
        result,
        response,
        spy;

      beforeEachAsync(async function() {
        data = {
          spring: jasmine.createSpy('spring'),
          servers: [],
          serverSpark: {}
        };

        response = {
          objects: [
            {
              command_and_host: {
                command: {
                  cancelled: false,
                  complete: false,
                  created_at: '2014-12-10T17:11:00.852148+00:00',
                  dismissed: false,
                  errored: false,
                  id: '390',
                  jobs: ['/api/job/390/'],
                  logs: '',
                  message: 'Setting up host lotus-34vm5.iml.intel.com',
                  resource_uri: '/api/command/390/'
                },
                host: {
                  address: 'lotus-34vm5.iml.intel.com',
                  available_actions: [],
                  available_jobs: [],
                  available_transitions: [],
                  boot_time: null,
                  client_mounts: [],
                  content_type_id: 35,
                  corosync_reported_up: false,
                  fqdn: 'lotus-34vm5.iml.intel.com',
                  id: '32',
                  immutable_state: true,
                  install_method: 'existing_keys_choice',
                  label: 'lotus-34vm5.iml.intel.com',
                  locks: {
                    read: [],
                    write: [390]
                  },
                  member_of_active_filesystem: false,
                  needs_fence_reconfiguration: false,
                  needs_update: false,
                  nids: [],
                  nodename: 'lotus-34vm5',
                  private_key: null,
                  private_key_passphrase: null,
                  properties: '{}',
                  resource_uri: '/api/host/32/',
                  root_pw: null,
                  server_profile: {
                    default: false,
                    initial_state: 'unconfigured',
                    managed: false,
                    name: 'default',
                    resource_uri: '/api/server_profile/default/',
                    ui_description: 'An unconfigured server.',
                    ui_name: 'Unconfigured Server',
                    user_selectable: false,
                    worker: false
                  },
                  state: 'undeployed',
                  state_modified_at: '2014-12-10T17:11:00.845748+00:00',
                  version: null
                }
              },
              error: null,
              traceback: null
            },
            {
              command_and_host: {
                command: {
                  cancelled: false,
                  complete: false,
                  created_at: '2014-12-10T17:11:03.280882+00:00',
                  dismissed: false,
                  errored: false,
                  id: '391',
                  jobs: ['/api/job/391/'],
                  logs: '',
                  message: 'Setting up host lotus-34vm6.iml.intel.com',
                  resource_uri: '/api/command/391/'
                },
                host: {
                  address: 'lotus-34vm6.iml.intel.com',
                  available_actions: [],
                  available_jobs: [],
                  available_transitions: [],
                  boot_time: null,
                  client_mounts: [],
                  content_type_id: 35,
                  corosync_reported_up: false,
                  fqdn: 'lotus-34vm6.iml.intel.com',
                  id: '33',
                  immutable_state: true,
                  install_method: 'existing_keys_choice',
                  label: 'lotus-34vm6.iml.intel.com',
                  locks: {
                    read: [],
                    write: [391]
                  },
                  member_of_active_filesystem: false,
                  needs_fence_reconfiguration: false,
                  needs_update: false,
                  nids: [],
                  nodename: 'lotus-34vm6',
                  private_key: null,
                  private_key_passphrase: null,
                  properties: '{}',
                  resource_uri: '/api/host/33/',
                  root_pw: null,
                  server_profile: {
                    default: false,
                    initial_state: 'unconfigured',
                    managed: false,
                    name: 'default',
                    resource_uri: '/api/server_profile/default/',
                    ui_description: 'An unconfigured server.',
                    ui_name: 'Unconfigured Server',
                    user_selectable: false,
                    worker: false
                  },
                  state: 'undeployed',
                  state_modified_at: '2014-12-10T17:11:03.273059+00:00',
                  version: null
                }
              },
              error: null,
              traceback: null
            },
            {
              command_and_host: {
                command: false,
                host: {
                  address: 'lotus-34vm7.iml.intel.com',
                  available_actions: [],
                  available_jobs: [],
                  available_transitions: [],
                  boot_time: null,
                  client_mounts: [],
                  content_type_id: 35,
                  corosync_reported_up: false,
                  fqdn: 'lotus-34vm7.iml.intel.com',
                  id: '33',
                  immutable_state: true,
                  install_method: 'existing_keys_choice',
                  label: 'lotus-34vm7.iml.intel.com',
                  locks: {
                    read: [],
                    write: [391]
                  },
                  member_of_active_filesystem: false,
                  needs_fence_reconfiguration: false,
                  needs_update: false,
                  nids: [],
                  nodename: 'lotus-34vm7',
                  private_key: null,
                  private_key_passphrase: null,
                  properties: '{}',
                  resource_uri: '/api/host/34/',
                  root_pw: null,
                  server_profile: {
                    default: false,
                    initial_state: 'unconfigured',
                    managed: false,
                    name: 'default',
                    resource_uri: '/api/server_profile/default/',
                    ui_description: 'An unconfigured server.',
                    ui_name: 'Unconfigured Server',
                    user_selectable: false,
                    worker: false
                  },
                  state: 'undeployed',
                  state_modified_at: '2014-12-10T17:11:03.273059+00:00',
                  version: null
                }
              },
              error: null,
              traceback: null
            }
          ]
        };

        createOrUpdateHostsStream = jasmine
          .createSpy('createOrUpdateHostsStream')
          .and.returnValue(highland([response]));

        waitForCommandCompletion = jasmine
          .createSpy('waitForCommandCompletion')
          .and.callFake(val => highland([val]));

        getHostProfiles = jasmine.createSpy('getHostProfiles').and.returnValue(
          highland([
            {
              some: 'profiles'
            }
          ])
        );

        onEnter = selectServerProfileStep.onEnter;

        result = onEnter(
          data,
          createOrUpdateHostsStream,
          getHostProfiles,
          waitForCommandCompletion,
          true
        );

        spy = jasmine.createSpy('spy');

        const stream = await result.hostProfileStream;
        stream.each(spy);
      });

      it('should create or update the hosts', () => {
        expect(createOrUpdateHostsStream).toHaveBeenCalledOnceWith(
          data.servers
        );
      });

      it('should pass commands to wait for command completion', () => {
        const commands = fp.flow(
          x => x.objects,
          fp.map(x => x.command_and_host),
          fp.map(x => x.command),
          fp.filter(x => x)
        )(response);

        expect(waitForCommandCompletion).toHaveBeenCalledOnceWith(
          true,
          commands
        );
      });

      it('should call getHostProfiles', () => {
        const hosts = fp.map(x => x.command_and_host.host, response.objects);

        expect(getHostProfiles).toHaveBeenCalledOnceWith(data.spring, hosts);
      });

      it('should return data and a hostProfileStream', () => {
        expect(result).toEqual({
          data: data,
          hostProfileStream: jasmine.any(Promise)
        });
      });

      it('should return the profiles', () => {
        expect(spy).toHaveBeenCalledOnceWith({ some: 'profiles' });
      });
    });
  });
});
