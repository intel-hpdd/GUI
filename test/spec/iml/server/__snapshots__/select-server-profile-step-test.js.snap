// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`select server profile select server profile step ctrl profile stream should display in the correct order 1`] = `
Array [
  Object {
    "hosts": Array [
      Object {
        "address": "lotus-34vm5.iml.intel.com",
        "invalid": false,
        "problems": Array [],
        "uiName": "Monitored Storage Server",
      },
      Object {
        "address": "lotus-34vm6.iml.intel.com",
        "invalid": false,
        "problems": Array [],
        "uiName": "Monitored Storage Server",
      },
    ],
    "invalid": false,
    "name": "base_monitored",
    "uiName": "Monitored Storage Server",
  },
  Object {
    "hosts": Array [
      Object {
        "address": "lotus-34vm5.iml.intel.com",
        "invalid": true,
        "problems": Array [
          Object {
            "description": "ZFS is installed but is unsupported by the Managed Storage Server profile",
            "error": "Result unavailable while host agent starts",
            "pass": false,
            "test": "zfs_installed == False",
          },
        ],
        "uiName": "Managed Storage Server",
      },
      Object {
        "address": "lotus-34vm6.iml.intel.com",
        "invalid": true,
        "problems": Array [
          Object {
            "description": "ZFS is installed but is unsupported by the Managed Storage Server profile",
            "error": "Result unavailable while host agent starts",
            "pass": false,
            "test": "zfs_installed == False",
          },
        ],
        "uiName": "Managed Storage Server",
      },
    ],
    "invalid": false,
    "name": "base_managed",
    "uiName": "Managed Storage Server",
  },
  Object {
    "hosts": Array [
      Object {
        "address": "lotus-34vm5.iml.intel.com",
        "invalid": true,
        "problems": Array [
          Object {
            "description": "The profile is designed for version 7 of EL",
            "error": "",
            "pass": false,
            "test": "distro_version < 8 and distro_version >= 7",
          },
        ],
        "uiName": "Managed Storage Server For EL7.2",
      },
      Object {
        "address": "lotus-34vm6.iml.intel.com",
        "invalid": true,
        "problems": Array [
          Object {
            "description": "The profile is designed for version 7 of EL",
            "error": "",
            "pass": false,
            "test": "distro_version < 8 and distro_version >= 7",
          },
        ],
        "uiName": "Managed Storage Server For EL7.2",
      },
    ],
    "invalid": false,
    "name": "base_managed_rh7",
    "uiName": "Managed Storage Server For EL7.2",
  },
  Object {
    "hosts": Array [
      Object {
        "address": "lotus-34vm5.iml.intel.com",
        "invalid": false,
        "problems": Array [],
        "uiName": "POSIX HSM Agent Node",
      },
      Object {
        "address": "lotus-34vm6.iml.intel.com",
        "invalid": false,
        "problems": Array [],
        "uiName": "POSIX HSM Agent Node",
      },
    ],
    "invalid": false,
    "name": "posix_copytool_worker",
    "uiName": "POSIX HSM Agent Node",
  },
  Object {
    "hosts": Array [
      Object {
        "address": "lotus-34vm5.iml.intel.com",
        "invalid": false,
        "problems": Array [],
        "uiName": "Robinhood Policy Engine Server",
      },
      Object {
        "address": "lotus-34vm6.iml.intel.com",
        "invalid": false,
        "problems": Array [],
        "uiName": "Robinhood Policy Engine Server",
      },
    ],
    "invalid": false,
    "name": "robinhood_server",
    "uiName": "Robinhood Policy Engine Server",
  },
]
`;

exports[`select server profile select server profile step ctrl profile stream with invalid profiles should display in the correct order 1`] = `
Array [
  Object {
    "hosts": Array [
      Object {
        "address": "lotus-34vm5.iml.intel.com",
        "invalid": false,
        "problems": Array [],
        "uiName": "Monitored Storage Server",
      },
      Object {
        "address": "lotus-34vm6.iml.intel.com",
        "invalid": false,
        "problems": Array [],
        "uiName": "Monitored Storage Server",
      },
    ],
    "invalid": false,
    "name": "base_monitored",
    "uiName": "Monitored Storage Server",
  },
  Object {
    "hosts": Array [
      Object {
        "address": "lotus-34vm5.iml.intel.com",
        "invalid": false,
        "problems": Array [],
        "uiName": "POSIX HSM Agent Node",
      },
      Object {
        "address": "lotus-34vm6.iml.intel.com",
        "invalid": false,
        "problems": Array [],
        "uiName": "POSIX HSM Agent Node",
      },
    ],
    "invalid": false,
    "name": "posix_copytool_worker",
    "uiName": "POSIX HSM Agent Node",
  },
  Object {
    "hosts": Array [
      Object {
        "address": "lotus-34vm5.iml.intel.com",
        "invalid": false,
        "problems": Array [],
        "uiName": "Robinhood Policy Engine Server",
      },
      Object {
        "address": "lotus-34vm6.iml.intel.com",
        "invalid": false,
        "problems": Array [],
        "uiName": "Robinhood Policy Engine Server",
      },
    ],
    "invalid": false,
    "name": "robinhood_server",
    "uiName": "Robinhood Policy Engine Server",
  },
  Object {
    "hosts": Array [
      Object {
        "address": "lotus-34vm5.iml.intel.com",
        "invalid": true,
        "problems": Array [
          Object {
            "description": "ZFS is installed but is unsupported by the Managed Storage Server profile",
            "error": "Result unavailable while host agent starts",
            "pass": false,
            "test": "zfs_installed == False",
          },
        ],
        "uiName": "Managed Storage Server",
      },
      Object {
        "address": "lotus-34vm6.iml.intel.com",
        "invalid": true,
        "problems": Array [
          Object {
            "description": "ZFS is installed but is unsupported by the Managed Storage Server profile",
            "error": "Result unavailable while host agent starts",
            "pass": false,
            "test": "zfs_installed == False",
          },
        ],
        "uiName": "Managed Storage Server",
      },
    ],
    "invalid": true,
    "name": "base_managed",
    "uiName": "Managed Storage Server",
  },
  Object {
    "hosts": Array [
      Object {
        "address": "lotus-34vm5.iml.intel.com",
        "invalid": true,
        "problems": Array [
          Object {
            "description": "The profile is designed for version 7 of EL",
            "error": "",
            "pass": false,
            "test": "distro_version < 8 and distro_version >= 7",
          },
        ],
        "uiName": "Managed Storage Server For EL7.2",
      },
      Object {
        "address": "lotus-34vm6.iml.intel.com",
        "invalid": true,
        "problems": Array [
          Object {
            "description": "The profile is designed for version 7 of EL",
            "error": "",
            "pass": false,
            "test": "distro_version < 8 and distro_version >= 7",
          },
        ],
        "uiName": "Managed Storage Server For EL7.2",
      },
    ],
    "invalid": true,
    "name": "base_managed_rh7",
    "uiName": "Managed Storage Server For EL7.2",
  },
]
`;
